### PR TITLE
Clean repos with only spotlight texture projection files 

### PR DESCRIPTION
**Pull request by lostink**
Cleaned the whole repos and only replace those files we have modified.

Added texture projection for spot light.

### CODE REVIEW

**Comment by sebavan** [comment]
This looks better in number of files (only a couple left).

Could you remove the branching in the shader and replace by #ifdef ? 

I ll review more in depth the changes after the holidays, have a good thanksgiving.

**Comment by deltakosh** [comment]
Hello team:) any news?

File: src/Shaders/ShadersInclude/lightsFragmentFunctions.fx [hunk 1]
```diff
@@ -112,3 +109,12 @@ lightingInfo computeHemisphericLighting(vec3 viewDirectionW, vec3 vNormal, vec4
 
 	return result;
 }
+
+lightingInfo computeProjectionTexture(lightingInfo origin,sampler2D projectionLightSampler, mat4 textureProjectionMatrix){
+	lightingInfo result = origin;
+	vec4 strq = textureProjectionMatrix * vec4(vPositionW, 1.0);
+	strq /= strq.w;
+	vec4 textureColor = texture2D(projectionLightSampler, vec2(strq[0], strq[1]));
```
Comment (sebavan) [inline src/Shaders/ShadersInclude/lightsFragmentFunctions.fx]: replace vec2(strq[0], strq[1])); by strq.xy

File: src/Shaders/ShadersInclude/lightsFragmentFunctions.fx [hunk 2]
```diff
@@ -76,7 +73,7 @@ lightingInfo computeSpotLighting(vec3 viewDirectionW, vec3 vNormal, vec4 lightDa
 
 		result.specular = specComp * specularColor * attenuation;
 #endif
-		return result;
+        return result;
```
Comment (sebavan) [inline src/Shaders/ShadersInclude/lightsFragmentFunctions.fx]: do not change ident if not necessary

File: src/Materials/babylon.materialHelper.ts [hunk 3]
```diff
@@ -264,13 +270,17 @@
...
                     "depthValues" + lightIndex,
+                    "textureProjectionMatrix" + lightIndex,
                 );
 
                 if (uniformBuffersList) {
                     uniformBuffersList.push("Light" + lightIndex);
                 }
-
-                samplersList.push("shadowSampler" + lightIndex);
+                //====================
```
Comment (sebavan) [inline src/Materials/babylon.materialHelper.ts]: remove comment

File: src/Lights/babylon.spotLight.ts [hunk 4]
```diff
@@ -123,6 +214,12 @@
                 normalizeDirection.z,
                 Math.cos(this.angle * 0.5),
                 lightIndex);
+
+            //=======================================
```
Comment (sebavan) [inline src/Lights/babylon.spotLight.ts]: remove comment

File: src/Lights/babylon.spotLight.ts [hunk 5]
```diff
@@ -25,9 +99,26 @@
...
+            return this._textureProjectionMatrix;
+        }
+        /**
+        * Allows setting the value of projection texture
+        */
+        public set textureMatrix(value: Matrix) {
+            this._textureProjectionMatrix = value;
+        }
+
+        //=================||=====================================
```
Comment (sebavan) [inline src/Lights/babylon.spotLight.ts]: remove comment

File: src/Lights/babylon.spotLight.ts [hunk 6]
```diff
@@ -25,9 +99,26 @@
             this._shadowAngleScale = value;
             this.forceProjectionMatrixCompute();
         }
-
         @serialize()
         public exponent: number;
+
+        //========================================================
```
Comment (sebavan) [inline src/Lights/babylon.spotLight.ts]: remove comment

File: src/Lights/babylon.spotLight.ts [hunk 7]
```diff
@@ -1,5 +1,79 @@
...
+            Warning:
+            Change the angle of the Spotlight, direction of the SpotLight will not re-compute the 
+            projection matrix. Need to call computeTextureMatrix() to recompute manually. Add inheritance
+            to the setting function of the 2 attributes will solve the problem.
+        */
+        /**
+         * Main function for light texture projection matrix computing.
+         */
+        public computeTextureMatrix(): void{    
+            /*
```
Comment (sebavan) [inline src/Lights/babylon.spotLight.ts]: remove comment if code in is not usefull

File: src/Lights/babylon.spotLight.ts [hunk 8]
```diff
@@ -1,5 +1,79 @@
...
+
+            Warning:
+            Change the angle of the Spotlight, direction of the SpotLight will not re-compute the 
+            projection matrix. Need to call computeTextureMatrix() to recompute manually. Add inheritance
+            to the setting function of the 2 attributes will solve the problem.
+        */
+        /**
+         * Main function for light texture projection matrix computing.
+         */
+        public computeTextureMatrix(): void{
```
Comment (sebavan) [inline src/Lights/babylon.spotLight.ts]: Could this be protected instead ?

File: src/Lights/babylon.light.ts [hunk 9]
```diff
@@ -384,7 +384,57 @@
...
+        /**
+         * Allows reading the near clip of the Spotlight for texture projection.
+         */
+        public get light_near(): number {
+            return this._light_near;
+        }
+        /**
+         * Allows setting the near clip of the Spotlight for texture projection.
+         */
+        public set light_near(value: number) {
```
Comment (sebavan) [inline src/Lights/babylon.light.ts]: this should force computing the matrix if it changes.

File: src/Lights/babylon.light.ts [hunk 10]
```diff
@@ -384,7 +384,57 @@
...
+         * Allows reading the far clip of the Spotlight for texture projection.
+         */
+        public get light_far(): number {
+            return this._light_far;
+        }
+        /**
+         * Allows setting the far clip of the Spotlight for texture projection.
+         */
+        public set light_far(value: number) {
+            this._light_far = value;
```
Comment (sebavan) [inline src/Lights/babylon.light.ts]: this should force computing the matrix if it changes

**Comment by deltakosh** [comment]
up?

**Comment by lostink** [comment]
Hi. It seems that we have some local merging problem right now. We have changed those parts and we thought we have pushed to the github but actually not. Working on the merging part now.

**Comment by lostink** [comment]
Hello,

We have fixed our local/remote repos pushing and merging problem and had
pushed the newest branch to the pull request. Could you review again on
that please so if anything need to be changed we can continue to work on
that.

Thank you!

Jiahao & Yalun

2018-01-04 17:57 GMT-05:00 David Catuhe <notifications@github.com>:

**Comment by deltakosh** [comment]
You have a few errors raised by Travis

**Comment by deltakosh** [comment]
There is something wrong with this PR:) it changes 737 files :)

**Comment by lostink** [comment]
I will close this one and re-open a pull request for merging the newest version of code.

**Pull request closed without merging**